st2api/st2api/controllers/resource.py

```diff
@@ -171,92 +187,87 @@ def _get_all(self, exclude_fields=None, sort=None, offset=0, limit=None, query_o
+        return self._get_one_by_id(id=id, exclude_fields=exclude_fields, **kwargs)
 
-    def _get_one_by_id(self, id, exclude_fields=None, from_model_kwargs=None):
+    def _get_one_by_id(self, id, exclude_fields=None, from_model_kwargs=None,
+                       requester_user=None, permission_type=None, **kwargs):
         """
         :param exclude_fields: A list of object fields to exclude.
         :type exclude_fields: ``list``
         """
 
-        LOG.info('GET %s with id=%s', pecan.request.path, id)
-
         instance = self._get_by_id(resource_id=id, exclude_fields=exclude_fields)
 
+        rbac_utils.assert_user_has_resource_db_permission(user_db=requester_user,
+                                                          resource_db=instance,
+                                                          permission_type=permission_type)
+
         if not instance:
             msg = 'Unable to identify resource with id "%s".' % id
-            pecan.abort(http_client.NOT_FOUND, msg)
+            abort(http_client.NOT_FOUND, msg)
 
         from_model_kwargs = from_model_kwargs or {}
-        from_model_kwargs.update(self._get_from_model_kwargs_for_request(request=pecan.request))
+        from_model_kwargs.update(self._get_from_model_kwargs_for_request(**kwargs))
         result = self.model.from_model(instance, **from_model_kwargs)
-        LOG.debug('GET %s with id=%s, client_result=%s', pecan.request.path, id, result)
 
         return result
 
-    def _get_one_by_name_or_id(self, name_or_id, exclude_fields=None, from_model_kwargs=None):
+    def _get_one_by_name_or_id(self, name_or_id, exclude_fields=None, from_model_kwargs=None,
+                               requester_user=None, permission_type=None, **kwargs):
         """
         :param exclude_fields: A list of object fields to exclude.
         :type exclude_fields: ``list``
         """
 
-        LOG.info('GET %s with name_or_id=%s', pecan.request.path, name_or_id)
-
         instance = self._get_by_name_or_id(name_or_id=name_or_id, exclude_fields=exclude_fields)
 
+        if permission_type:
+            rbac_utils.assert_user_has_resource_db_permission(user_db=requester_user,
+                                                              resource_db=instance,
+                                                              permission_type=permission_type)
+
         if not instance:
             msg = 'Unable to identify resource with name_or_id "%s".' % (name_or_id)
-            pecan.abort(http_client.NOT_FOUND, msg)
+            abort(http_client.NOT_FOUND, msg)
 
         from_model_kwargs = from_model_kwargs or {}
-        from_model_kwargs.update(self._get_from_model_kwargs_for_request(request=pecan.request))
+        from_model_kwargs.update(self._get_from_model_kwargs_for_request(**kwargs))
         result = self.model.from_model(instance, **from_model_kwargs)
-        LOG.debug('GET %s with name_or_id=%s, client_result=%s', pecan.request.path, id, result)
```

Comment:
The reason I removed this particular line is that it uses pecan.request.path which can't be exposed through the means of openapi. Saying that, yes, we do have logging middleware https://github.com/StackStorm/st2/pull/2727/files/d802d6afbe760a4e05944214389d16616d963331#diff-f745fc7cee78f050f04933589c1924e9